src/gobuploadservice/storage/storage_handler.py

```diff
@@ -0,0 +1,154 @@
+    return wrapper_decorator
+
+
+class GOBStorageHandler():
+    """Metadata aware Storage handler """
+    EVENT_TABLE = "event"
+
+    def __init__(self, gob_metadata):
+        """Initialize DBHandler with gob metadata
+
+        This will create abstractions to entities and events, and initialize storage if necessary
+
+        """
+        self.metadata = gob_metadata
+        self.engine = create_engine(URL(**GOB_DB))
+        self._get_reflected_base()
+        self._init_storage()
+        self.session = None
+
+    def _get_reflected_base(self):
+        self.base = automap_base()
+        self.base.prepare(self.engine, reflect=True)
+        self.base.metadata.reflect(bind=self.engine)
+
+    def _init_storage(self):
+        if not hasattr(self.base.classes, self.metadata.entity):
+            # Create events table if not yet exists
+            if not hasattr(self.base.classes, self.EVENT_TABLE):
+                self._init_event()
+
+            # create table
+            self._init_entity()
+
+            # refresh reflected base
+            self._get_reflected_base()
+
+        if self.metadata.version != "0.1":
+            # No migrations defined yet...
+            raise ValueError("Unexpected version, please write a generic migration here of migrate the import")
+
+    def _init_event(self):
+        """
+        Create the events table
+        """
+        meta = MetaData(self.engine)
+
+        columns = [get_column(column) for column in EVENTS.items()]
+        table = Table(self.EVENT_TABLE, meta, *columns, extend_existing=True)
+        table.create(self.engine, checkfirst=True)
+
+    def _init_entity(self):
+        """
+        Initialize a database table for the given metadata.
+        """
+
+        table_name = self.metadata.entity    # e.g. meetbouten
+        model = self.metadata.model          # the GOB model for the specified entity
+
+        # internal columns
+        columns = [get_column(column) for column in FIXED_COLUMNS.items()]
+        columns.extend([get_column(column) for column in METADATA_COLUMNS['private'].items()])
+
+        # externally visible columns
+        columns.extend([get_column(column) for column in METADATA_COLUMNS['public'].items()])
+
+        # get the entity columns
+        data_column_desc = {col: desc['type'] for col, desc in model.items()}
+        columns.extend([get_column(column) for column in data_column_desc.items()])
+
+        # Create an index on source and source_id for performant updates
+        index = Index(f"{table_name}.idx.source_source_id", "_source", "_source_id", unique=True)
+
+        meta = MetaData(self.engine)
+
+        table = Table(table_name, meta, *columns, index, extend_existing=True)
+        table.create(self.engine, checkfirst=True)
+
+    @property
+    def DbEvent(self):
+        return self.base.classes.event
+
+    @property
+    def DbEntity(self):
+        return getattr(self.base.classes, self.metadata.entity)
+
```

Comment:
db or storage? I would recommend consistent naming.
If the storage handler encapsulates the storage (being a db, file, json, collections of csv's) the name db is a little confusing. Actually, the data can be stored wherever what structure. Following this reasoning, why not give the storage handler simply properties event and entity?